proxy/http2/Http2ConnectionState.cc

```diff
@@ -567,6 +567,11 @@ rcv_settings_frame(Http2ConnectionState &cstate, const Http2Frame &frame)
     }
   }
 
+  if (frame.is_from_early_data()) {
```

Comment:
We cannot receive SETTINGS frame from early data?

Reply:
For now I'm only allowing a SETTINGS frame if the `ACK` is set, since settings frame is used for changing settings so I thought we don't want to allow it over early data.

Reply:
I understand your concern but a client cannot send HEADERS before sending own SETTINGS if I understand the spec correctly. 

H2 spec says:
> A SETTINGS frame MUST be sent by both endpoints at the start of a connection and MAY be sent at any other time by either endpoint over the lifetime of the connection.

My understanding is that SETTINGS in this paragraph is one with setting parameters but not one with ACK flag on. So if we prohibited receiving SETTINGS without ACK it would mean we can only receive the preface from early data. I presume changing settings is a part of connection setup and it should not be limited.

Reply:
Actually not really, I've tested with ATS and as long as there is a SETTINGS frame, it doesn't matter what it contains, and according to spec, the SETTINGS frame that follows PREFACE could be empty.

> That is, the connection preface starts with the string PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n). This sequence MUST be followed by a SETTINGS frame (Section 6.5), which MAY be empty.

Reply:
Well, yeah, SETTINGS frame may be empty, but we still need the empty frame. I meant a SETTINGS frame that is not for ACK.